src/test/java/BaseTest.java

```diff
@@ -1,11 +1,116 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+
+import java.time.Duration;
+
 
 public class BaseTest {
 
+    public WebDriver driver;
+    private WebDriverWait wait;
+
+    public WebElement notificationMsg;
+
+    public String url = "https://qa.koel.app";
+
+
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
-}
\ No newline at end of file
+
+    @BeforeMethod
+    @Parameters({"BaseUrl"})
+    public void launchBrowser(String BaseUrl) {
+        //Added ChromeOptions argument to fix websocket error
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+
+        //Open Chrome Browser
+        driver = new ChromeDriver(options);
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
+        driver.manage().window().maximize();
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(5));
```

Comment:
Hello,
Your homework 20 has been reviewed, but the test script failed.  Please recheck and correct it.
Best wishes from the HW team (Khanh, khanh@testpro.io)

![image](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/144403954/d92df1c5-4740-4c13-9958-d9c7fb7f997a)

//HINT: At the beginning of the `BaseTest `class you have declared the wait variable `private WebDriverWait wait;` line 20. So When function  lanunchBrower you do not need to initialize the wait variable anymore `WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(5));`

Need to reuse variable ` wait = new WebDriverWait(driver, Duration.ofSeconds(5));`